engine/src/main/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionProcessor.java

```diff
@@ -89,12 +101,21 @@ public ProcessingError tryHandleError(
 
   private void tryDeleteResources(final TypedRecord<ResourceDeletionRecord> command) {
     final var value = command.getValue();
+
+    final var processOptional =
+        Optional.ofNullable(processState.getProcessByKey(value.getResourceKey()));
+    if (processOptional.isPresent()) {
+      deleteProcess(processOptional.get());
+      return;
+    }
```

Comment:
🔧 Optional should typically only be used as the return type of a method, not to build flow control.

```java
// This is simpler than using Optional.ofNullable
final var process = processState.getProcessByKey(value.getResourceKey());
if (process != null) {
  deleteProcess(process);
  return;
}
```

Reply:
Ugh I don't like null checks. Ideally the `getProcessByKey` would just return the `Optional` but that is too much refactoring.